core/uvisor.h

```diff
@@ -52,13 +52,22 @@
 /* unprivileged box as called by privileged code */
 typedef void (*UnprivilegedBoxEntry)(void);
 
+UVISOR_FORCEINLINE void uvisor_noreturn(void)
+{
+    volatile int var = 1;
+    while(var);
```

Comment:
Consider marking this function as no return with `UVISOR_NORETURN`. This will allow the compiler to reason about the function and its use better.

Consider also using `for(;;)` instead of a spin on a volatile variable.

Reply:
Why is there a preference to using `for(;;)` instead?
Using this method it will be easier to debug, you just need to change the variable on-the-fly to continue (instead of changing the code and re-compile).

Reply:
One can also just change the PC.